js/oceanex.js

```diff
@@ -0,0 +1,617 @@
+                        'trades',
+                        'timestamp',
+                    ],
+                },
+                'private': {
+                    'get': [
+                        'key',
+                        'members/me',
+                        'orders',
+                        'orders/filter',
+                    ],
+                    'post': [
+                        'orders',
+                        'orders/multi',
+                        'order/delete',
+                        'order/delete/multi',
+                        'orders/clear',
+                    ],
+                },
+            },
+            'version': 'v1',
+            'rateLimit': 3000,
+            'has': {
+                'fetchMarkets': true,
+                'fetchCurrencies': false,
+                'fetchTicker': true,
+                'fetchOrderBook': true,
+                'fetchTrades': true,
+                'fetchTradingLimits': false,
+                'fetchTradingFees': true,
+                'fetchFundingFees': false,
+                'fetchTime': true,
+                'fetchOrder': true,
+                'fetchOrders': false,
+                'fetchBalance': true,
+                'createMarketOrder': false,
+                'createOrder': true,
+                'cancelOrder': true,
+                'cancelAllOrders': true,
+            },
+            'timeframes': {
+                '1m': '1m',
+                '5m': '5m',
+                '15m': '15m',
+                '30m': '30m',
+                '1h': '1h',
+                '4h': '4h',
+                '12h': '12h',
+                '1d': '1d',
+                '1w': '1w',
+            },
+            'requiredCredentials': {
+                'apiKey': true,
+                'secret': false,
+            },
+            'exceptions': {
+                'codes': {
+                    '-2': BadRequest,
+                    '1001': BadRequest,
+                    '1006': AuthenticationError,
+                    '1011': PermissionDenied,
+                    '2002': InvalidOrder,
+                    '2004': OrderNotFound,
+                    '9003': PermissionDenied,
+                },
+                'exact': {
+                    'market does not have a valid value': BadRequest,
+                    'side does not have a valid value': BadRequest,
+                    'Account::AccountError: Cannot lock funds': InsufficientFunds,
+                    'The account does not exist': AuthenticationError,
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetMarkets (this.extend ({ 'show_details': true }));
+        let result = [];
+        const markets = this.safeValue (response, 'data');
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            let id = this.safeValue (market, 'id').toUpperCase ();
```

Comment:
Let's remove `.toUppercase ()` here ↑
And then let's remove `market['id'].toLowerCase ()` everywhere. That will save us some cycles.